massa-network/src/peer_info_database.rs

```diff
@@ -616,34 +537,32 @@ impl PeerInfoDatabase {
     /// # Argument
     /// * ip : ip address of the considered peer.
     pub fn out_connection_closed(&mut self, ip: &IpAddr) -> Result<(), NetworkError> {
+        let peer_type = self.get_peer_type(ip).ok_or({
+            NetworkError::PeerConnectionError(NetworkConnectionErrorType::PeerInfoNotFoundError(
+                *ip,
+            ))
+        })?;
+
+        self.update_global_active_out_connection_count(
+            peer_type,
+            false,
+            NetworkConnectionErrorType::CloseConnectionWithNoConnectionToClose(*ip),
+        )?;
+
         let peer = self.peers.get_mut(ip).ok_or({
```

Comment:
it's always a bit srange to do get_peer_type and then get the whole peer with get_mut. Why not get_mut the peer once, and just check its type as a first step to improve performance ? (also elsewhere in the code)

Reply:
because when we update the global counters, we take mutable ownership of self. PeerType is Copy so getting peer type does not borrow self. Whereas taking the peer mutably once takes mutable ownership of self, and then I cannot update the global counters.

Reply:
okay